waspc/packages/wasp-config/src/mapUserSpecToAppSpecDecls.ts

```diff
@@ -64,60 +71,39 @@ export function mapUserSpecToDecls(
       emailSender,
       websocket
     ),
-  })
-
-  for (const [actionName, actionConfig] of actions.entries()) {
-    decls.push({
-      declType: 'Action',
-      declName: actionName,
-      declValue: mapOperationConfig(actionConfig, parseEntityRef),
-    })
-  }
-
-  for (const [queryName, queryConfig] of queries.entries()) {
-    decls.push({
-      declType: 'Query',
-      declName: queryName,
-      declValue: mapOperationConfig(queryConfig, parseEntityRef),
-    })
   }
 
-  for (const [apiName, apiConfig] of apis.entries()) {
-    decls.push({
-      declType: 'Api',
-      declName: apiName,
-      declValue: mapApiConfig(apiConfig, parseEntityRef),
-    })
-  }
-
-  for (const [jobName, jobConfig] of jobs.entries()) {
-    decls.push({
-      declType: 'Job',
-      declName: jobName,
-      declValue: mapJob(jobConfig, parseEntityRef),
-    })
-  }
-
-  for (const [
-    apiNamespaceName,
-    apiNamespaceConfig,
-  ] of apiNamespaces.entries()) {
-    decls.push({
-      declType: 'ApiNamespace',
-      declName: apiNamespaceName,
-      declValue: mapApiNamespace(apiNamespaceConfig),
-    })
-  }
+  return makeDeclsArray({
+    App: [appDecl],
+    Page: pageDecls,
+    Route: routeDecls,
+    Action: actionDecls,
+    Query: queryDecls,
+    Api: apiDecls,
+    Job: jobDecls,
+    ApiNamespace: apiNamespaceDecls,
+    Crud: crudDecls,
+  })
+}
 
-  for (const [crudName, crudConfig] of spec.cruds.entries()) {
-    decls.push({
-      declType: 'Crud',
-      declName: crudName,
-      declValue: mapCrud(crudConfig, parseEntityRef),
-    })
-  }
+function makeDeclsArray(decls: {
+  [Type in AppSpec.Decl['declType']]: AppSpec.GetDeclForType<Type>[]
+}): AppSpec.Decl[] {
+  return Object.values(decls).flatMap((decl) => [...decl])
+}
 
-  return decls
+function mapToDecls<T, const DeclType extends AppSpec.Decl['declType']>(
```

Comment:
What does `const` do here?

Reply:
Nothing, it's a leftover I think. Nice catch.